express-notes-api/index.js

```diff
@@ -0,0 +1,85 @@
+const express = require('express');
+const app = express();
+const fs = require('fs');
+const notesJSON = require('./data.json');
+
+app.use('/', express.json());
+
+function rewriteNotes(notes) {
+  const updatedNotes = JSON.stringify(notes, null, 2);
+  fs.writeFile('data.json', updatedNotes, 'utf-8', function (err, res) {
+    if (err) {
+      res.status(500).json({ error: 'An unexpected error occurred.' });
+      process.exit(1);
+    }
+  });
```

Comment:
If possible, avoid shutting down your server with `throw` or `process.exit()`. Even if there is an error writing the new data, it should still at least be possible to read the data instead of just going offline.

I recommend just inlining your calls to `fs.writeFile` directly in the routes that need it. That way you can conditionally respond with either `500` or some success code